lua/autorun/nl_library.lua

```diff
@@ -0,0 +1,607 @@
+// The tazer is a dependancy at this point //
+AllowedWake = { // these conditions can be removed when a friend presses +use
+	["ZZZ"] = true,
+	["---"] = true,
+	["PLV"] = false
+}
+
+NLL = {}
+
+//NetworkStrings//
+if SERVER then
+	util.AddNetworkString("nllsendhealth")
+	util.AddNetworkString("nllplayercond")
+	util.AddNetworkString("nllupdatebuddha")
+	util.AddNetworkString("nllsetdrowsy")
+
+net.Receive("nllupdatebuddha",function(len,ply)
+	ply.CovertBuddha = net.ReadInt(4)
+end)
+
+// Block speaking while sleeping or stunned
+hook.Add("PlayerSay", "TG_NLL", function(ply, str)
+	if ply.nllismuted then return "" end
+end)
+hook.Add("PlayerCanSeePlayersChat", "TG_NLL", function(text, teamOnly, listener, talker)
+	if talker.nllismuted then
+		return false
+	end
+end)
+hook.Add("PlayerCanHearPlayersVoice", "TG_NLL", function(listener, talker)
+	if talker.nllismuted then
+		return false,false
+	end
+end)
+
+// Handles updating the displayed health on bodies.
+hook.Add("Think", "TG_NLL", function()
+	for k,v in pairs(player.GetAll()) do
```

Comment:
What stops you from using `EntityTakeDamage` or similar to find out if a player has taken damage instead of iterating through each player each tick?